app/modules/key_management/secrets_schema.py

```
@@ -39,8 +46,14 @@ def validate_provider_and_api_key(cls, provider: str, values):
         elif provider == "deepseek":
             if not api_key.startswith("sk-or-"):
                 raise ValueError("Invalid OpenRouter API key format")
-        else:
-            raise ValueError("Invalid provider")
+        elif provider == "mistralai": # Assuming mistralai also uses openrouter keys
+            if not api_key.startswith("sk-or-"):
+                raise ValueError("Invalid OpenRouter API key format for MistralAI")
+        elif provider == "meta-llama": # Assuming meta-llama also uses openrouter keys
+            if not api_key.startswith("sk-or-"):
```

Ruff (0.8.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)